### ISSUE TITLE
Yank to selection is broken with the new nim native messenger.

### ISSUE THREAD

**Issue by dlobue** [tridactyl/tridactyl#3616]
-   Brief description of the problem:
Yank to selection is broken with the new nim native messenger.
Interestingly, paste from selection does work with the nim native messenger.

I've confirmed the issue is specific to the nim native messenger by modifying `~/.mozilla/native-messaging-hosts/tridactyl.json` to point at the python native messenger, and yank immediately starts working again.

-   Steps to reproduce:

    1. `:set yankto selection`
    2. browse to any website
    3. `yy`
    4. note no message that the url has been yanked is shown
    5. check selection buffer contents and note they were not updated.

-   Tridactyl version (`:version`):

Tridactyl version: 1.21.1pre5614-69340235

-   Firefox version (Top right menu > Help > About Firefox):

Firefox version: Mozilla Firefox 87.0

-   URL of the website the bug happens on:

Every website.

-   Config (in a new tab, run `:viewconfig --user`, copy the url and paste it somewhere like gist.github.com):

https://gist.github.com/dlobue/363dd6a8f686cfd18dd2df9ff1c315e6

-   Contents of ~/.tridactylrc or ~/.config/tridactyl/tridactylrc (if they exist):

```
" Add helper commands that Mozillians think make Firefox irredeemably
" insecure. For details, read the comment at the top of this file.
command fixamo_quiet jsb tri.excmds.setpref("privacy.resistFingerprinting.block_mozAddonManager", "true").then(tri.excmds.setpref("extensions.webextensions.restrictedDomains", '""'))
command fixamo js tri.excmds.setpref("privacy.resistFingerprinting.block_mozAddonManager", "true").then(tri.excmds.setpref("extensions.webextensions.restrictedDomains", '""').then(tri.excmds.fillcmdline_tmp(3000, "Permissions added to user.js. Please restart Firefox to make them take affect.")))

" Make Tridactyl work on more sites at the expense of some security. For
" details, read the comment at the top of this file.
fixamo_quiet

autocmd DocStart nomnoml.com mode ignore
autocmd DocStart scalyr.com/.script mode ignore
autocmd TabEnter .* unfocus

seturl notion.so preventautofocusjackhammer true

set yankto selection
set putfrom selection
set relatedopenpos next
set tabopenpos next
set theme dark
set visualenterauto false
set allowautofocus false
set smoothscroll false
set tabopenpos next
set relatedopenpos next
set csp clobber


bind + zoom 0.1 true
bind - zoom -0.1 true

bind yg composite js document.location.href.replace(/https?:\/\//,"git@").replace("/",":").replace(/\/?$/,".git") | clipboard yank

bind j scrollline 3
bind k scrollline -3
```

-   Operating system:
Operating system: linux

-   Result of running `:! echo $PATH`, or `! echo %PATH%` on Windows, in Tridactyl:

/home/dominic/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/bin 

-   Unix-like only:
    -   `:native` less than 0.2.0: result of running `printf '%c\0\0\0{"cmd": "run", "command": "echo $PATH"}' 39 | ~/.local/share/tridactyl/native_main.py` in a terminal:

    {"cmd": "run", "content": "/home/dominic/.zinit/polaris/bin:/home/dominic/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/go/site/bin:/usr/share/git/remote-helpers:/home/dominic/.go/bin:/home/dominic/.rvm/bin:/home/dominic/.jenv/bin:/home/dominic/.pyenv/bin:/home/dominic/bin:/home/dominic/.local/bin:/home/dominic/.node_modules/bin:/home/dominic/.npm/bin:/home/dominic/node_modules/.bin:/home/dominic/.krew/bin:./node_modules/.bin:/home/dominic/.gem/ruby/2.7.0/bin\n", "code": 0}

    -   `:native` at least version 0.2.0: result of running `printf '%c\0\0\0{"cmd": "run", "command": "echo $PATH"}' 39 | ~/.local/share/tridactyl/native_main` in a terminal:

    {"cmd":"run","content":"/home/dominic/.zinit/polaris/bin:/home/dominic/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/go/site/bin:/usr/share/git/remote-helpers:/home/dominic/.go/bin:/home/dominic/.rvm/bin:/home/dominic/.jenv/bin:/home/dominic/.pyenv/bin:/home/dominic/bin:/home/dominic/.local/bin:/home/dominic/.node_modules/bin:/home/dominic/.npm/bin:/home/dominic/node_modules/.bin:/home/dominic/.krew/bin:./node_modules/.bin:/home/dominic/.gem/ruby/2.7.0/bin\n","command":"echo $PATH","code":0}

-   Other notes:

I've tried multiple versions of the nim native messenger: 0.2.5, 0.3.3, and the latest 0.3.5

-   Output of `:native`

    Native messenger is correctly installed, version 0.3.5

**Comment by dlobue** [comment]
Oh, one other note: yank to clipboard seems to work without problems in the nim native messenger. I'm using that setting with nim native messenger 0.2.5 on my work mac without any problems.

**Comment by bovine3dom** [comment]
Reproducible on Linux - thanks for reporting : )

(FWIW, "yank to clipboard" doesn't use the native messenger, so one wouldn't expect it to be broken)

**Comment by bovine3dom** [comment]
Fixed in https://github.com/tridactyl/native_messenger/pull/60 by pvs. Will probably be released in native 0.3.7 whenever I remember to do that :)

**Comment by whirm** [comment]
I'd appreciate if you released 0.3.7 as I think I'm hitting this same bug.

Thanks!
